pkg/providers/v1/aws.go

```diff
@@ -1621,7 +1710,7 @@ func parseMetadataLocalHostname(metadata string) (string, []string) {
 }
 
 // extractNodeAddresses maps the instance information from EC2 to an array of NodeAddresses
-func extractNodeAddresses(instance *ec2.Instance) ([]v1.NodeAddress, error) {
+func extractIPv4NodeAddresses(instance *ec2.Instance) ([]v1.NodeAddress, error) {
```

Comment:
Can we add a comment or rename the variables to make it clear that privateDNSName and publicDNSName are ipv4 specific?

Reply:
Something like this?